pyproject.toml

```
@@ -1,3 +1,69 @@
+[tool.poetry]
+name = "rdflib"
+version = "6.3.0a0"
+description = """
+RDFLib is a Python library for working with RDF, \
+a simple yet powerful language for representing information."""
+authors = ["Daniel 'eikeon' Krech <eikeon@eikeon.com>"]
+maintainers = ["RDFLib Team <rdflib-dev@googlegroups.com>"]
+repository = "https://github.com/RDFLib/rdflib"
+documentation = "https://rdflib.readthedocs.org/"
+license = "BSD-3-Clause"
+classifiers=[
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "License :: OSI Approved :: BSD License",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Operating System :: OS Independent",
+    "Natural Language :: English"
+]
+readme = "README.md"
+
+[tool.poetry.scripts]
+rdfpipe = 'rdflib.tools.rdfpipe:main'
+csv2rdf = 'rdflib.tools.csv2rdf:main'
+rdf2dot = 'rdflib.tools.rdf2dot:main'
+rdfs2dot = 'rdflib.tools.rdfs2dot:main'
+rdfgraphisomorphism = 'rdflib.tools.graphisomorphism:main'
+
+[tool.poetry.dependencies]
+python = "^3.7"
+isodate = "^0.6.1"
+pyparsing = "^3.0.9"
+#setuptools  # not needed anymore?
+importlib_metadata = {version = "^4.2.0", python = ">=3.7,<3.8"}
+berkeleydb = {version = "^18.1.5", optional = true}
+networkx = [
+    {version = "^2.6.2", python = ">3.7,<3.8", optional = true},
+    {version = "^2.8.8", python = "^3.8", optional = true},
+]
+
+[tool.poetry.group.dev.dependencies]
+html5lib = "^1.1"
```

So that it can be used in `[tool.poetry.extras]`

```suggestion
networkx = [
    {version = "^2.6.2", python = ">3.7,<3.8", optional = true},
    {version = "^2.8.8", python = "^3.8", optional = true},
]
html5lib = "^1.1"

[tool.poetry.group.dev.dependencies]
```

---

Actually this is somewhat wrong, but the essence of it is that we should try and retain the same extras as before, and we used to have `html` installing `html5lib` so would be best to keep it like that.

---

Got it. I didn't understand this request at first but now I do - I've made the update in the most recent commit.  Note that in order to make it so that html5lib is only installed as an "extra", we need to specify it with `optional = true`.  I've done that in my commit, and also added the necessary line to the `extras` section.

Looks like I've omitted some other previously-configured `extra` groups too - I'll need to address those, unless we decide they shouldn't be extras after all.

---

I would say the only extras that we should definitely keep are the one for normal users:

```
{
"html": ["html5lib"],
"berkeleydb": ["berkeleydb"],
"networkx": ["networkx"],
}
```

The others, `tests`, `docs`, `dev` are less important to keep as extras think.

Currently readthedocs uses the `docs` extras:

https://github.com/RDFLib/rdflib/blob/8213b7465783e636a394c86fd2387a44523d9c2f/.readthedocs.yaml#L19-L24


But I think we should find a way to get readthedocs to use the poetry.lock file so the builds are more reproducible, I will look at that.

There are also this issue:

- https://github.com/RDFLib/rdflib/issues/1519#issuecomment-999191547

I'm not entirely sure of the workflow here but it seems there is some expectation of having `tests` extras maybe, however the right solution here could just be to fix the tests that use optional dependencies to only run when those are installed.

I would be fine if we don't keep extras for `docs`, `dev` and `tests`. We can always add it back if it causes serious problems.

---

CC: @kloczek you may have some input that is relevant here also.

---

@aucampia, looks like this snippet provides a useful example of how to use poetry with readthedocs.yml: https://github.com/readthedocs/readthedocs.org/issues/4912#issuecomment-1143587902

---

resolving this as I'm fairly happy with what you did, will though make a further suggestion in a PR to your branch.